pepper-apis/src/main/java/org/broadinstitute/ddp/client/Auth0ManagementClient.java

```diff
@@ -399,33 +443,84 @@ public User removeUserGuidForAuth0User(String auth0UserId, String auth0ClientId)
+    /**
+     * Initiate password reset flow and get the ticket URL by auth0 user id.
+     *
+     * @param auth0UserId the user to do password reset for
+     * @param redirectUrl the URL to go to after user has finished password reset
+     * @return result with ticket URL, or error response
+     */
+    public ApiResult<String, APIException> createPasswordResetTicket(String auth0UserId, String redirectUrl) {
+        String msg = "Hit rate limit while creating password reset ticket for auth0 user " + auth0UserId + ", retrying";
+        var ticket = new PasswordChangeTicket(auth0UserId);
+        ticket.setResultUrl(redirectUrl);
+        return tryPasswordChangeRequest(ticket, msg);
+    }
+
+    /**
+     * Initiate password reset flow and get the ticket URL by email and connection.
+     *
+     * @param email        the user's email
+     * @param connectionId the auth0 connection id
+     * @param redirectUrl  the URL to go to after user has finished password reset
+     * @return result with ticket URL, or error response
+     */
+    public ApiResult<String, APIException> createPasswordResetTicket(String email, String connectionId, String redirectUrl) {
+        String msg = String.format(
+                "Hit rate limit while creating password reset ticket for email %s and connection %s, retrying",
+                email, connectionId);
+        var ticket = new PasswordChangeTicket(email, connectionId);
+        ticket.setResultUrl(redirectUrl);
+        return tryPasswordChangeRequest(ticket, msg);
+    }
+
+    private ApiResult<String, APIException> tryPasswordChangeRequest(PasswordChangeTicket ticket, String retryMessage) {
+        return withRetries(retryMessage, () -> {
+            try {
+                mgmtApi.setApiToken(getToken());
+                PasswordChangeTicket createdTicket = mgmtApi.tickets().requestPasswordChange(ticket).execute();
+                return ApiResult.ok(200, createdTicket.getTicket());
+            } catch (APIException e) {
+                return ApiResult.err(e.getStatusCode(), e);
+            } catch (Exception e) {
+                return ApiResult.thrown(e);
+            }
+        });
+    }
+
+    private <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResult<B, E>> callback) {
         ApiResult<B, E> res = null;
-        while (numTries > 0) {
+        int numTries = 0;
+        int maxTries = maxRetries + 1;
+        while (numTries < maxTries) {
             res = callback.get();
+            numTries++;
+            if (numTries >= maxTries) {
+                break;
+            }
             if (res.getStatusCode() == 429) {
-                LOG.error(errorMsg, res.getError());
+                LOG.error(retryMessage, res.getError());
+                long wait = backoffMillis * numTries + new Random().nextInt(MAX_JITTER_MILLIS);
```

Comment:
Previously, we waited a set amount of time. Now, we're doing more of a "exponential backoff". There's no exponent here but it will help space things out more.